### PR TITLE
Assignment for week 12

### PR DESCRIPTION
**Pull request by morarodgers**
Added all CRUD operations

### CODE REVIEW

File: public/notes.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            message.textContent = "You have been logged off.";
+            notesTable.replaceChildren([notesTableHeader]);
+
+            showLoginRegister();
+        } else if (e.target.classList.contains("editButton")) {
+            message.textContent = "";
+            showAddEdit(e.target.dataset.id);
+        } else if (e.target.classList.contains("deleteButton")) {
+          message.textContent = "";
+          showAddEdit(e.target.dataset.id);
```
Comment (joseluiz) [inline public/notes.js]: When we click the delete button, the application behaves as if we had clicked the edit button. Therefore, we need to create a method that calls the endpoint to delete the note and replace the current action of calling the showAddEdit method with this new method.

File: public/addEdit.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      } else if (e.target === editCancel) {
+        message.textContent = "";
+        
+        showNotes();
+
+      } else if (e.target.classList.contains("deleteButton")) {
+        // Get the noteId from the data attribute
+        const noteId = e.target.dataset.id;
+
+        try {
```
Comment (joseluiz) [inline public/addEdit.js:84]: You can create a separate method called deleteNote and call it in notes.js line 37 when the delete button is pressed. Then, you can remove this logic from here.

**Review by joseluiz — CHANGES_REQUESTED** [review]
Hey @morarodgers. I have left two comments regarding the issues that need to be fixed. The problem is related to the delete note functionality which is currently not working. Please check the comments in the PR for more details. If you have any questions, please feel free to contact me on Slack.

File: public/notes.js (hunk 1 above)
Comment (morarodgers) [reply public/notes.js]: Hello Jose
I have managed to add a function to delete
Kindly confirm if I have done it right.

**Review by joseluiz — APPROVED** [review]
Hey @morarodgers , everything is okay now. Good job!

**Pull request merged**